### ISSUE TITLE
Percentages mishandled in `color(lab L a b)`

### ISSUE THREAD

**Issue by svgeesus** [LeaVerou/color.js#70]
In `color(lab L a b)`, same as `lab(L a b)`, the L parameter [is required to have a percentage](https://drafts.csswg.org/css-color-4/#valdef-color-lab) and 50% [resolves to an L value of 50, not 0.5](https://drafts.csswg.org/css-color-4/#specifying-lab-lch).

However, color.js has two errors:

a) it allows the "%" to be omitted, for example[`color(lab 54.903 -54.15 55.313)`](
https://colorjs.io/apps/convert/?color=color(lab%2054.903%20-54.15%2055.313)&precision=5)

b) if the percent is used, 100% resolves to an L of 1, not 100. For example [`color(lab 54.903% -54.15 55.313)`](https://colorjs.io/apps/convert/?color=color(lab%2054.903%25%20-54.15%2055.313)&precision=5) is a near-black.

Note that [`lab(54.903% -54.15 55.313)`](https://colorjs.io/apps/convert/?color=lab(54.903%25%20-54.15%2055.313)&precision=5) is handled correctly. Although [`lab(54.903 -54.15 55.313)`](https://colorjs.io/apps/convert/?color=lab(54.903%20-54.15%2055.313)&precision=5) is also accepted, and treated as identical to the percentage form.

**Comment by svgeesus** [comment]
I suspect the problem is [here](https://github.com/LeaVerou/color.js/blob/master/src/color.js#L672)

```js
if (/%$/.test(arg)) {
	// Convert percentages to 0-1 numbers
	let n = new Number(+arg.slice(0, -1) / 100);
	n.percentage = true;
	return n;
}
```

but then that would also affect the `lab()` and `lch()` functions. Ah, they have [special handling](https://github.com/LeaVerou/color.js/blob/fae51efd108eda3a80ea18ccda890b9f2d702891/src/spaces/lab.js#L55-L60):

```js
	let L = parsed.args[0];

	// Percentages in lab() don't translate to a 0-1 range, but a 0-100 range
	if (L.percentage) {
		parsed.args[0] = L * 100;
	}
```

so that same test for "is the first argument CIE L" needs to be applied to the general `color()` parsing in src/color.js

**Comment by svgeesus** [comment]
Confirming [bug still exists](https://colorjs.io/apps/convert/?color=color(lab%2050%25%200%200)&precision=4)

color(lab 50% 0 0) → color(lab 0.5 0 0)

**Comment by LeaVerou** [comment]
Since `color(lab L a b)` is not valid CSS anymore, perhaps we should just not accept it…

**Comment by svgeesus** [comment]
Agreed, provided `lab(number number number)` and `lab(percent number number)` are handled correctly with L=100% mapping to L=100 and vice-versa.

(There is also `lab(percent percent percent)` from CSS, but that is unrelated to this issue)

**Comment by LeaVerou** [comment]
Reopening because we *do* still accept it 😛

**Comment by svgeesus** [comment]
> Note that [`lab(54.903% -54.15 55.313)`](https://colorjs.io/apps/convert/?color=lab(54.903%25%20-54.15%2055.313)&precision=5) is handled correctly. Although [`lab(54.903 -54.15 55.313)`](https://colorjs.io/apps/convert/?color=lab(54.903%20-54.15%2055.313)&precision=5) is also accepted, and treated as identical to the percentage form.

Note that CSS Color 4 has changed and using a lightness without a percent is [now allowed](https://drafts.csswg.org/css-color-4/#specifying-lab-lch) with 0 being the same as 0% and 100 the same as 100%. So that part is now correct.

We don't accept percent for `a` and `b` though, which is also now allowed.

**Comment by svgeesus** [comment]
> Ah, they have [special handling](https://github.com/LeaVerou/color.js/blob/fae51efd108eda3a80ea18ccda890b9f2d702891/src/spaces/lab.js#L55-L60):

The reference ranges for all color spaces that accept `<percentage>` should be part of the color space definition, not special-cased into the code like this.
